client/src/components/campaignCreatorComponents/OptionListCampaignCreator.tsx

```diff
@@ -1,25 +1,40 @@
-import { Checkbox, Flex, Heading } from '@radix-ui/themes';
+import { Flex } from '@radix-ui/themes';
 import { TwilightCheckbox, TwilightNodeHeader } from '../primitiveComponents/Primitives';
+import { NodePillarLists, TypeCampaignData } from './CampaignTypeConfig';
+import campaignOptionsData from '../../static_data/campaign_creator.json';
 
 interface OptionListCampaignCreatorProps {
     header: string;
-    options: string[];
-    data: Record<string, boolean>;
-    setData: React.Dispatch<React.SetStateAction<{}>>;
+    optionKey: keyof NodePillarLists;
+    data: TypeCampaignData;
+    setData: React.Dispatch<React.SetStateAction<TypeCampaignData>>;
 }
 
-function OptionListCampaignCreator({ header, options, data, setData }: OptionListCampaignCreatorProps) {
+function OptionListCampaignCreator({ header, optionKey, data, setData }: OptionListCampaignCreatorProps) {
     const handlCheckedAction = (label: string, checked: boolean) => {
-        const temp: Record<string, boolean> = { ...data };
-        temp[label] = checked;
-        setData(temp);
-    };
+        const updatedData: TypeCampaignData = { ...data };
 
+        if (checked && !updatedData[optionKey].includes(label)) {
+            updatedData[optionKey] = [...updatedData[optionKey], label];
+        } else if (!checked && updatedData[optionKey].includes(label)) {
+            const modifiedList: string[] = updatedData[optionKey].filter((item) => item !== label);
+            updatedData[optionKey] = [...modifiedList];
```

Comment:
I'm thinking for now we should specify a sort of rule for "functional data".
Which is to say, I'm seeing some of the patterns but they aren't consistent and seem arbitrarily used.

I think a safe start to work around it without getting bogged in minutia is to only particularly put attention to immutability on non-locally-sourced vars (anything not defined as `const` within its own scope).  Then do whatever you want to local vars